package.json

```diff
@@ -165,6 +166,14 @@
     "webpack-cli": "^4.5.0",
     "yorkie": "^2.0.0"
   },
+  "peerDependencies": {
+    "jiti": "^1.21.0"
+  },
+  "peerDependenciesMeta": {
+    "jiti": {
+      "optional": true
+    }
+  },
```

Comment:
I'm not sure if the potentially breaking change of adding a peer dependency mentioned in the RFC is addressed.

I tried using this version in a test project that uses an out-of-range version of jiti:

```
"devDependencies": {
  "eslint": "github:aryaemami59/eslint#eslint.config.ts",
  "jiti": "1.20.0"
}
```

But `npm install` fails due to "Conflicting peer dependency":

```
$ npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: eslint@9.7.0
npm ERR! Found: jiti@1.20.0
npm ERR! node_modules/jiti
npm ERR!   dev jiti@"1.20.0" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peerOptional jiti@"^1.21.0" from eslint@9.7.0
npm ERR! node_modules/eslint
npm ERR!   dev eslint@"github:aryaemami59/eslint#eslint.config.ts" from the root project
npm ERR!   peer eslint@"^6.0.0 || ^7.0.0 || >=8.0.0" from @eslint-community/eslint-utils@4.4.0
npm ERR!   node_modules/@eslint-community/eslint-utils
npm ERR!     @eslint-community/eslint-utils@"^4.2.0" from eslint@9.7.0
npm ERR!
npm ERR! Conflicting peer dependency: jiti@1.21.6
npm ERR! node_modules/jiti
npm ERR!   peerOptional jiti@"^1.21.0" from eslint@9.7.0
npm ERR!   node_modules/eslint
npm ERR!     dev eslint@"github:aryaemami59/eslint#eslint.config.ts" from the root project
npm ERR!     peer eslint@"^6.0.0 || ^7.0.0 || >=8.0.0" from @eslint-community/eslint-utils@4.4.0
npm ERR!     node_modules/@eslint-community/eslint-utils
npm ERR!       @eslint-community/eslint-utils@"^4.2.0" from eslint@9.7.0
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

Reply:
Maybe we could remove the `peerDependencies` from `package.json` to avoid the error during `npm install`. In `loadFlatConfigFile()`, after importing `jiti`, we could check that `jitiModule?.createJiti` is a function and throw an error if it is not. That would ensure that jiti 2 is being used (beta or stable).

Reply:
I think we still could add it as an optional peer dependency like this:

```json
  "peerDependencies": {
    "jiti": "*"
  },
  "peerDependenciesMeta": {
    "jiti": {
      "optional": true
    }
  },
```

And then throw an error if `jitiModule?.createJiti` is not a function like you said. Technically we don't have to throw an error since `jiti` will still work as long as we fallback to `jitiModule.default` and the config file doesn't contain top-level `await` syntax.

Reply:
Indeed, that's the only way to make it potentially nonbreaking.

However, if someone has an invalid version of jiti installed, eslint will still start throwing an error with that approach, won't it?

Reply:
I think that's the best way to approach this. We could do something like what `@typescript-eslint` does [here](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-estree/src/parseSettings/warnAboutTSVersion.ts) as was previously mentioned in [this comment](https://github.com/eslint/rfcs/pull/117#issuecomment-2120851994). That way we make sure to avoid a breaking change.

Reply:
Throwing when the user has an old jiti version definitely sounds like a breaking change to me. Why not just act like there's no jiti when it doesn't have the expected function rather than throwing?

And at that point, I think it would be reasonable to remove the peer dependency completely. A peer dependency that's optional and with range `*` is pretty meaningless. It means you can have any version installed, or none, which is true of every package. The only effect would be to cause bogus problems for package managers that don't support peerDependenciesMeta.

Reply:
When jiti is outdated, we would only throw an error if the user attempts to use a TypeScript config file in place of a JavaScript file and also only if they pass the `unstable_ts_config` flag. Neither is possible so far, so that shouldn't be a breaking change. I think the error would be helpful to let users know why their TypeScript config file doesn't work as expected and how that can be fixed.

Reply:
Ah, yes that makes sense, in that case I agree that shouldn't be considered breaking. But still, a peer dependency that's optional and with range * doesn't seem value-add to me, but not a big deal since most package managers respect peerDependenciesMeta now as far as I know.

Reply:
> But still, a peer dependency that's optional and with range * doesn't seem value-add

I'm not sure what's the best practice for our use case, but I believe not declaring at all (as any kind of production dependency)  a package we use in runtime makes it a "phantom dependency" which might be problematic for some package managers.

Reply:
jiti v1 has forward compatible `jiti.import()` API (with less features but exact same API) exactly for the same reason of solving possible dependency overlap issues.